### PR TITLE
Fixed #27868 -- Filtered locale path subdirectories

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/i18n/test_extraction.py [hunk 1]
```diff
@@ -604,13 +604,26 @@ def test_keep_pot_enabled(self):
...
+        """
+        When the `locale` flag is absent, all dirs from the parent locale dir
+        are considered as language directories, except if the directory doesn't
+        start with two letters (which excludes __pycache__, .gitignore, etc.).
+        """
+        os.mkdir(os.path.join('locale', '_do_not_pick'))
+        # Excluding locales that do not compile
+        management.call_command('makemessages', exclude=['ja', 'es_AR'], verbosity=0)
+        self.assertTrue(os.path.exists(self.PO_FILE_KO))
+        self.assertFalse(os.path.exists('locale/_do_not_pick/LC_MESSAGES/django.po'))
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: I wonder if something like `self.PO_FILE_KO.replace('/ko/', '_do_not_pick`)` would make that a bit more resilient to future changes. No strong feeling either way.

File: tests/i18n/test_extraction.py [hunk 2]
```diff
@@ -604,13 +604,26 @@ def test_keep_pot_enabled(self):
...
     def test_multiple_locales(self):
         management.call_command('makemessages', locale=['pt', 'de'], verbosity=0)
         self.assertTrue(os.path.exists(self.PO_FILE_PT))
         self.assertTrue(os.path.exists(self.PO_FILE_DE))
 
+    def test_all_locales(self):
+        """
+        When the `locale` flag is absent, all dirs from the parent locale dir
+        are considered as language directories, except if the directory doesn't
+        start with two letters (which excludes __pycache__, .gitignore, etc.).
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: "two or three" ? -- based on `{2,3}`?

Comment (claudep) [inline tests/i18n/test_extraction.py]: In fact, I don't think checking the third character is bringing anything at all, I'll fix the regex instead.

File: tests/i18n/test_extraction.py (hunk 1 above)
Comment (claudep) [inline tests/i18n/test_extraction.py]: Not convinced :-)

**Pull request merged**